src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -688,15 +695,58 @@ private static void addColumnToSchemaMutation(TableMetadata table, ColumnMetadat
     {
         AbstractType<?> type = column.type;
         if (type instanceof ReversedType)
-            type = ((ReversedType) type).baseType;
+            type = ((ReversedType<?>) type).baseType;
 
-        builder.update(Columns)
+        Row.SimpleBuilder rowBuilder = builder.update(Columns)
                .row(table.name, column.name.toString())
                .add("column_name_bytes", column.name.bytes)
                .add("kind", column.kind.toString().toLowerCase())
                .add("position", column.position())
                .add("clustering_order", column.clusteringOrder().toString().toLowerCase())
                .add("type", type.asCQL3Type().toString());
+
+        // Dynamic data masking functions shouldn't be attached to columns during rolling upgrades
+        // to avoid sending mutations with columns that are unknown to the old nodes.
+        ColumnMask mask = column.getMask();
+        if (ColumnMask.clusterSupportsMaskedColumns())
```

Comment:
As I am reading the PR I wonder if `clusterSupportsMaskedColumns()` should be moved to `QueryProcessor` or some other 'higher'/clusterish class as being buried in `ColumnMask` sounds weird to me.

Reply:
I don't know what that class would be. `QueryProcessor` is more CQL related and more related to the coordinator than to the cluster, so I don't see it there. Maybe in `Gossiper`, since the method calls it twice? But that would couple `Gossiper` with masking, which is not ideal.